gflownet/proxy/crystals/composition.py

```diff
@@ -0,0 +1,73 @@
+import pickle
+from pathlib import Path
+
+import numpy as np
+import torch
+from torchtyping import TensorType
+
+from gflownet.proxy.base import Proxy
+
+PROTONS_NUMBER_COUNTS = None
+
+
+def _read_protons_number_counts():
+    global PROTONS_NUMBER_COUNTS
+    if PROTONS_NUMBER_COUNTS is None:
+        with open(
+            Path(__file__).parent / "number_of_protons_counts.pkl", "rb"
+        ) as handle:
+            PROTONS_NUMBER_COUNTS = pickle.load(handle)
+    return PROTONS_NUMBER_COUNTS
+
+
+class CompositionMPFrequency(Proxy):
+    def __init__(self, normalise: bool = True, **kwargs):
+        super().__init__(**kwargs)
+        self.counts_dict = _read_protons_number_counts()
+        self.normalise = normalise
+
+    def _get_max_protons_state(self, env):
+        max_protons_state = env.source.copy()
+        for elem in env.required_elements:
+            max_protons_state[env.elem2idx[elem]] = max(1, env.min_atom_i)
+        for idx in range(env.min_diff_elem - len(env.required_elements)):
+            max_protons_state[-1 - idx] = max(1, env.min_atom_i)
+
+        for idx in range(len(max_protons_state)):
```

Comment:
I think it would be easier to read if instead of iterating over the indices of the list (`idx in range(len(max_protons_state))` and then indexing the list (e.g. `if max_protons_state[-1 - idx] == 0`, we iterate over the list directly (starting from the end, which is what seems to be needed here): `for el in max_protons_state[::-1]` and then simply `if el == 0`, etc.

Reply:
The problem is that I need idx to increment integers in the list `max_protons_state`. If I iterate over it like `for el in max_protons_state[::-1]` and work with `el`, the element of the list won't be changed as integers are immutable. Leaving it as it is for now (looks like it works according to the tests), but feel free to change that if you see a better way to code it

Reply:
Well, if you need the index, the standard way is then `for idx, el in enumerate(max_protons_state[::-1])`